### PR TITLE
[HLSL] move rcp to cgbuiltins

### LINKED ISSUE #87784: remove rcp intrinsic and move lowering out of DXILIntrinsicExpansion.cpp and into  CGBuiltin.cpp
SPIRV and DXIL both lower to `1/x`  so an intrinsic isn't necessary.

_Originally posted by @farzonl in https://github.com/llvm/llvm-project/issues/87367#issuecomment-2039918413_

### PR DESCRIPTION
**Pull request by farzonl**
Removing the intrinsic because there is no opCodes for rcp in DXIL or SPIR-V.
Moving means we don't have to re-implement this  feature for each backend.

fixes #87784

### CODE REVIEW

File: llvm/test/CodeGen/SPIRV/hlsl-intrinsics/rcp.ll [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+
+; CHECK-DAG: %[[#vec2_const_ones_f64:]] = OpConstantComposite %[[#vec2_float_64:]] %[[#const_f64_1:]] %[[#const_f64_1:]]
+; CHECK-DAG: %[[#vec3_const_ones_f64:]] = OpConstantComposite %[[#vec3_float_64:]] %[[#const_f64_1:]] %[[#const_f64_1:]] %[[#const_f64_1:]]
+; CHECK-DAG: %[[#vec4_const_ones_f64:]] = OpConstantComposite %[[#vec4_float_64:]] %[[#const_f64_1:]] %[[#const_f64_1:]] %[[#const_f64_1:]] %[[#const_f64_1:]]
+
+
+define spir_func noundef half @test_rcp_half(half noundef %p0) #0 {
+entry:
+  ; CHECK: %[[#arg0:]] = OpFunctionParameter %[[#float_16:]]
+  ; CHECK: OpFDiv %[[#float_16:]] %[[#const_f16_1:]] %[[#arg0:]]
```
Comment (python3kgae) [inline llvm/test/CodeGen/SPIRV/hlsl-intrinsics/rcp.ll:35]: Should we have same test to generate dxil op 'rcp'?

Comment (farzonl) [reply llvm/test/CodeGen/SPIRV/hlsl-intrinsics/rcp.ll:35]: Maybe im doing it wrong but the dxil lowering didn’t change the ir in a noticeable way.

**Review by coopp — APPROVED** [review]
Looks good to me

File: llvm/test/CodeGen/SPIRV/hlsl-intrinsics/rcp.ll (hunk 1 above)
Comment (python3kgae) [reply llvm/test/CodeGen/SPIRV/hlsl-intrinsics/rcp.ll:35]: If we always generate 1/a, maybe we could deprecate dxil op 'rcp'.
fdiv is legal dxil :)

Comment (farzonl) [reply llvm/test/CodeGen/SPIRV/hlsl-intrinsics/rcp.ll:35]: We don't have a dxil op for rcp, just the hlsl intrinsic. So that makes it easy for us.

Comment (farzonl) [reply llvm/test/CodeGen/SPIRV/hlsl-intrinsics/rcp.ll:35]: The one thing that we need to do is dxc  turns this vector division into an elementwise divide, but we don't have that yet in the dixl lowering.

**Review by python3kgae — APPROVED** [review]

**Pull request merged**